packages/babel-plugin-proposal-import-wasm-source/test/fixtures/node-cjs/import-expression-dynamic-specifier/output.js

```diff
@@ -0,0 +1,3 @@
+"use strict";
+
+let promise = Promise.resolve().then(() => require("fs").promises.readFile(require.resolve(getSpecifier())).then(WebAssembly.compile));
```

Comment:
Are we using `Promise.resolve()` here intentionally?

Reply:
Yes, so that if `getSpecifier()` throws it results in a rejected promise rather than a thrown error.

Reply:
Oh actually, this is wrong

Reply:
![image](https://github.com/babel/babel/assets/30521560/848f5ad2-78ba-47cc-8e96-6c1c9fe62226)

I tried it in nodejs and it seems it throws.🤔

Reply:
`new Promise(() => require("fs").promises.readFile(require.resolve("")).then(WebAssembly.compile))`
Would this behave the same?

Reply:
These would be both ok:
```js
new Promise((r, j) => require("fs").promises.readFile(require.resolve("")).then(WebAssembly.compile).then(r, j));
Promise.resolve().then(() => require("fs").promises.readFile(require.resolve("")).then(WebAssembly.compile));
```

Btw, I realized that if in a dynamic import stringifying the specifier throws, that error must be thrown synchronously. I pushed [03e32f7](https://github.com/babel/babel/pull/15870/commits/03e32f7c2410ecd11a8487b58609eb95889c024d) to handle this case.